src/1.2/index.ts

```diff
@@ -0,0 +1,82 @@
+import { constants, createReadStream, createWriteStream, WriteStream } from 'fs'
+import { access } from 'fs/promises'
+import path from 'path'
+import csv from 'csvtojson'
+import { URL } from 'url'
+
+const __dirname = new URL('./', import.meta.url).pathname
+
+function write(stream: WriteStream, data: string, cb: (value: void | PromiseLike<void>) => void) {
+    if (!stream.write(data)) {
+        stream.once('drain', cb);
+    } else {
+        process.nextTick(cb);
+    }
+}
+
+function onErrorCallback(err: Error) {
+    console.error(err)
+    process.exit(1)
+}
+ 
+process.stdout.write('Enter filename from csv directory (default file1):')
+process.stdin.setEncoding('utf-8')
+process.stdin.on('data', async (data) => {
+    const incoming = data.toString().replace(/\n/g, '') || 'file1'
+    const csvPath = path.resolve(__dirname, `csv/${incoming}.csv`)
+
+
+    access(csvPath, constants.R_OK)
+        .then(async () => {
+            const readStream = createReadStream(csvPath, {
```

Comment:
Ok, it's working solution.

What I still advice is to take a look how it can work with pipeline. Now the complexity of this solution is in proper error handling and stream closing + monitoring drain event. Pipeline will do it under the hood.